flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/UserCallbackFactory.java

```diff
@@ -0,0 +1,9 @@
+package org.apache.flink.streaming.connectors.kafka;
+
+import org.apache.flink.annotation.Public;
+import org.apache.kafka.clients.producer.Callback;
+
+@Public
+public interface UserCallbackFactory<IN> {
```

Comment:
Maybe, we should reach a consensus on the issue before implementation -- especially when adding a public interface.